text/0000-lts-release-strategy.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: default
+title: Fabric LTS release strategy
+nav_order: 3
+---
+
+- Feature Name: Fabric LTS release strategy
+- Start Date: 2020-03-18
+- RFC PR:
+- Fabric Component:
+- Fabric Issue:
+
+# Summary
+[summary]: #summary
+
+Define Long-term support (LTS) release strategy for Hyperledger Fabric
+
+Long-term support definition from wikipedia.org:
+
+> **Long-term support (LTS)** is a product lifecycle management policy in which a stable release of computer software is maintained for a longer period of time than the standard edition.
+
+> **LTS** applies the tenets of reliability engineering to the software development process and software release life cycle. Long-term support extends the period of software maintenance; it also alters the type and frequency of software updates (patches) to reduce the risk, expense, and disruption of software deployment, while promoting the dependability of the software.
+
+# Motivation
+[motivation]: #motivation
+
+Administrators of production deployments find LTS releases desirable as they provide confidence that a certain software release will be maintained for an extended period of time.
+
+Specifically, for Hyperledger Fabric, administrators can expect the following of a LTS release:
+
+- Stable and well-tested code
+- API contract stability
+- 3rd digit patch releases for bug and security fixes, delivered for an extended period of time on a regular cadence (or more frequently as needed)
+- Minimal set of feature additions and other changes that can easily be applied, reducing the risk of functional regressions and bugs
```

Comment:
This comment is applicable to the ledger package. As specified here, we did move a certain minimal set of features to v14 from v2. Though it was minimal, the code changes were moderate to significant including the logics. As a result, it was quite hard at times to move back and forth between v14 and v2, taking additional notes, etc... was bit painful. In this regard, my question is that why can't we just push bug fixes alone to LTS? I am not saying that we cannot manage multiple code versions but if it is required, we will prepare for it accordingly. When a risk is associated with both simple or complex feature, maybe we can restrict to bugs. If required, on a case by case, we can pick the most important minimalistic feature from master to LTS.

Reply:
We intend to only backport bug fixes to LTS releases. The last bullet is just a caveat that something small may get backported. For example if operators needed an additional metric in order to operate the LTS release effectively, we wouldn't necessarily block such a small change.

Reply:
IMO, calling a `metric addition` as a feature is misleading. If we intend to backport only bug fixes, I am not sure whey we have this confusing bullet. Anyway, this is just my opinion and feel free to ignore it.